var/spack/repos/builtin/packages/py-pysam/package.py

```diff
@@ -27,8 +27,17 @@ class PyPysam(PythonPackage):
     depends_on("py-cython@0.17:", type="build")
     depends_on("curl")
     depends_on("xz")
+    depends_on("htslib@:1.6", when="@:0.13")
+    depends_on("htslib")
 
     def setup_build_environment(self, env):
         env.set("LDFLAGS", self.spec["curl"].libs.search_flags)
+        # this flag is supposed to be removed by cy_build.py, but for some reason isn't
         if self.spec.platform == "darwin":
             env.remove_flags("LDSHARED", "-bundle")
+        # linking htslib, see:
+        # http://pysam.readthedocs.org/en/latest/installation.html#external
+        # https://github.com/pysam-developers/pysam/blob/v0.9.0/setup.py#L79
+        env.set("HTSLIB_MODE", "external")
+        env.set("HTSLIB_LIBRARY_DIR", self.spec["htslib"].prefix.lib)
+        env.set("HTSLIB_INCLUDE_DIR", self.spec["htslib"].prefix.include)
```

Comment:
```suggestion
        env.set("HTSLIB_LIBRARY_DIR", self.spec["htslib"].libs.directories[0])
        env.set("HTSLIB_INCLUDE_DIR", self.spec["htslib"].headers.directories[0])
```
On some systems lib64 is used instead of lib, this should support both.

Reply:
I tried this and got a build error:

```
==> Installing py-pysam-0.19.1-g37uykesacdxvmjwik25ptdqsfd35v2o
==> No binary for py-pysam-0.19.1-g37uykesacdxvmjwik25ptdqsfd35v2o found: installing from source
==> Error: NoLibrariesError: Unable to recursively locate htslib libraries in /Users/luke/Documents/GitHub/dantaslab/spack/opt/spack/darwin-monterey-m1/apple-clang-14.0.0/htslib-1.16-su7mfvegt5xjb7o7lzulziwcigvpna64

/Users/luke/Documents/GitHub/dantaslab/spack/var/spack/repos/builtin/packages/py-pysam/package.py:42, in setup_build_environment:
         39        # http://pysam.readthedocs.org/en/latest/installation.html#external
         40        # https://github.com/pysam-developers/pysam/blob/v0.9.0/setup.py#L79
         41        env.set("HTSLIB_MODE", "external")
  >>     42        env.set("HTSLIB_LIBRARY_DIR", self.spec["htslib"].libs.directories[0])
         43        env.set("HTSLIB_INCLUDE_DIR", self.spec["htslib"].headers.directories[0])
```

Reply:
You'll need to override the `libs` property in the htslib package. What is the library called? See zlib for an example where this is done.

Reply:
Do you mean add something like:

````
@property
def libs(self):
    return foo
````

to the `py-pysam` package or the `htslib` package?

Reply:
To the htslib package